### PR TITLE
[docs] README edits

### PR DESCRIPTION
**Pull request by edawson**
This adds two README changes.
1. The github checkout process is changed from git to https. This is recommended and also prevents users from having to associate their public key with their github account. However, I assume it was this way for a reason - please push back in code review and I can revert it.
2. Adds CMake as a listed dependency. I'd like to add `brew/apt` instructions for installing dependencies as well.

### CODE REVIEW

**Comment by edawson** [comment]
It looks like I might need to squash some commits here - I'll ping for help on this.

**Comment by tijyojwad** [comment]
@edawson sorry for taking time on this. Can you update this branch with changes from `dev-v0.5.0` as well? I think the only new change are the `https` link and the min cmake version

File: README.md [hunk 1]
```diff
@@ -43,15 +43,15 @@ This will clone the repo to the `master` branch, which contains code for latest
 and hot-fixes.
 
 ```
-git clone --recursive -b master git@github.com:clara-genomics/ClaraGenomicsAnalysis.git
+git clone --recursive -b master https://github.com/clara-genomics/ClaraGenomicsAnalysis.git
```
Comment (tijyojwad) [inline README.md:46]: @edawson can you fix this branch as well? it looks like it got some changes from `master` that have already been merged into `dev-v0.5.0`. rest of it looks good to me!

**Comment by tijyojwad** [comment]
Closes #470

**Comment by tijyojwad** [comment]
This is only a readme change, so no tests will be triggered. Force merging.

**Pull request merged**